agent/agent.go

```diff
@@ -1046,7 +1051,9 @@ func (a *Agent) consulConfig() (*consul.Config, error) {
 	base.Build = fmt.Sprintf("%s%s:%s", a.config.Version, a.config.VersionPrerelease, revision)
 
 	// Copy the TLS configuration
-	base.VerifyIncoming = a.config.VerifyIncoming || a.config.VerifyIncomingRPC
+	base.VerifyIncoming = a.config.VerifyIncoming
+	base.VerifyIncomingRPC = a.config.VerifyIncomingRPC
+
```

Comment:
🤔 this doesn't seem to be equivalent to the old logic. I'm sure if tests pass it's likely OK but having a hard time understanding why this changed?

Reply:
This changed because `tlsConfigurator` needs the vanilla config in order to be able to generate the different configs correctly. If `verify_incoming_rpc` is enabled and `verify_incoming_https` is disabled, `verify_incoming` would be still enabled for incoming HTTPS connections in this case. 🐛

Reply:
Well, I double checked and while my comment is true to some degree, it is also not quite correct. These changes are about the server config. Since the `tls.Configurator` is already initialised at that point and passed to `NewServerLogger`, it actually doesn't matter what is being set here. Which is why I reverted this change. 

I think eventually we should remove these lines to make sure they are not misleading.